### ISSUE TITLE
実験: 競りPPOのNapoleon失敗rewardを0にしてPASS collapseを再検証

### ISSUE THREAD

**Issue by eletim** [eletim/napoleon-web#364]
Related: #202, #359, #360, #362
Base branch: `feature/issue-202-non-playing-ai`

## 背景

Issue #362 では bidding-specific reward attribution を導入し、Citizen / Adjutant / All Pass / auction lost を bidding reward = 0 とした。

その結果、iter15では一時的に改善した。

- PASS rate: #360 92.1% -> #362 76.0%
- candidate Napoleon rate: #360 8.7% -> #362 26.0%

しかし iter30 では再び collapse した。

- PASS rate: 98.6%
- PASS top-1: 100%
- argmax bid: 0%
- PASS probability: 99.6%
- best bid probability: 0.2%
- PASS-bestBid gap: 6.169
- candidate Napoleon rate: 1.5%
- All Pass: 78.8%
- PASS advantage: +0.052
- BID advantage: -4.912

強いhandでも argmax bid は0%のままだった。

#362 により「他人の契約結果をPASSの正creditとして返す」問題は除去したが、bidしてNapoleonになった後の失敗reward `-5` が依然として強いnegative signalになっている可能性が高い。

次は reward attribution / opponent mix / PPO設定を維持し、Napoleon契約失敗時の bidding reward だけを `-5 -> 0` にして単独効果を検証する。

## 目的

bidding-specific reward v1 をベースに、candidate が Napoleon / Napoleon+Adjutant として契約に失敗した場合の bidding reward を0へ変更し、PASS collapseが改善するか確認する。

このIssueでは reward magnitude の1点だけを変更する。

## Bidding-specific reward v2

宣言枚数を `d` とする。

### candidate が Napoleon の場合

- win: `2 * d`
- lose: `0`

### candidate が Napoleon+Adjutant の場合

- win: `3 * d`
- lose: `0`

### candidate が契約を取らなかった場合

以下は従来どおり `0`。

- Citizen
- Adjutant
- All Pass
- auction lost
- candidate が途中でbidしたが最終的に別playerが契約を取った場合

## relative transform

#362 と同様、bidding rewardには game-mean relative transformを適用しない。

`bidding_training_reward_i = bidding_specific_raw_reward_i`

とする。

choosing-adjutant / exchange のreward設計は変更しない。

## この実験で確認したいこと

この変更は「PASSに罰を与える」ものではない。

弱いhandでPASSした場合も0、弱いhandでbidしてNapoleonになって失敗した場合も0になる。

一方、Napoleonとして成功したhandだけが大きな正rewardを受ける。

したがって、policyが

- 成功しやすいhandだけbidする方向を学べるのか
- penaltyが無いことで何でもbidする逆collapseへ進むのか
- それでもPASS top-1から抜けられないのか

を切り分ける。

## 維持する条件

loss reward以外は変更しない。

- #362 の bidding-specific reward attribution
- candidate x1 + Frozen x4 topology
- candidate seat 5席rotation
- Frozen bidding opponents: Conservative / All-Pass 50:50
- compact278 observation / model
- strongest-suit-balanced minibatch
- temperature=1 rollout sampling
- PPO clip epsilon
- entropy coefficient
- advantage normalization
- learning rate
- epochs
- batch size
- Frozen playing policy `ppo-separated-v1000`
- adjutant / exchange artifact
- game rule
- all-pass terminal handling

## provenance / resume safety

新しいrewardを別versionとして保存する。

最低限:

- bidding reward id/version
- Napoleon win=`2*d`
- Napoleon lose=`0`
- Napoleon+Adjutant win=`3*d`
- Napoleon+Adjutant lose=`0`
- non-contract outcomes=`0`
- bidding relative transform disabled

#362以前のrunを新rewardで誤resumeしないようfail-closeする。

## Unit test

固定fixtureで最低限以下を確認する。

### d=13

- Napoleon win = 26
- Napoleon lose = 0
- Napoleon+Adjutant win = 39
- Napoleon+Adjutant lose = 0
- Citizen = 0
- Adjutant = 0
- All Pass = 0
- auction lost = 0

### d=19

- Napoleon win = 38
- Napoleon lose = 0
- Napoleon+Adjutant win = 57
- Napoleon+Adjutant lose = 0

choosing-adjutant / exchange の既存rewardが変わっていないことも確認する。

## 短期再学習

まず30 iterationsで評価する。

Issue #362 と可能な限り同条件を使用する。

```text
iterations             30
games-per-iteration    200 logical seeds
epochs                  4
batch-size              128
learning-rate           1e-4
bidding entropy coef    0.1
advantage normalization none
minibatch strategy      strongest-suit-balanced
seed                    202
temperature             1
```

candidate 5-seat rotationにより actual games = 1000 / iteration を基本とする。

inference / training device は利用可能なbackendを使い、実験条件として記録する。
旧runは上書きしない。

## 必須diagnostics

iter 1 / 5 / 10 / 15 / 20 / 25 / 30 について最低限以下を記録する。

- PASS rate / bid rate
- PASS top-1 rate
- argmax bid rate
- PASS probability
- best legal bid probability
- PASS-bestBid gap
- policy entropy
- candidate Napoleon formation rate
- declaration success rate
- All Pass rate
- candidate role distribution
- PASS vs BID return
- PASS vs BID advantage
- successful Napoleon BID samples count / return / advantage
- failed Napoleon BID samples count / return / advantage

## Hand-strength / target diagnostics

Issue #362 より一段深く、最低限以下を分解する。

### hand-strength bin

- selected PASS rate
- PASS top-1
- argmax bid
- PASS probability
- best bid probability
- PASS-bestBid gap
- bid success rate
- return / advantage

### target 13..19

bid sampleについて:

- sample count
- Napoleon formation count
- success count / rate
- mean return
- mean advantage

可能なら strongest suit match / mismatch でも分解する。

## 比較対象

主比較は Issue #362 の30 iteration run。

iter30 baseline:

- PASS rate: 98.6%
- PASS top-1: 100%
- argmax bid: 0%
- PASS prob: 99.6%
- bestBid prob: 0.2%
- PASS-bestBid gap: 6.169
- candidate Napoleon rate: 1.5%
- All Pass: 78.8%
- declaration success: 0%
- PASS advantage: +0.052
- BID advantage: -4.912

Issue #360も参考値として残す。

## 成功判定

単にPASS率が下がれば成功とはしない。

望ましい方向:

- PASS top-1 100%から外れる
- argmax bidが明確に0%を超える
- 強いhandほどargmax bidが増える
- candidate Napoleon rateが回復する
- successful bidがpositive advantageを持ち、failed bidが過度なnegative advantageを持たなくなる
- stochastic PASS率が0%近辺へ逆collapseしない
- target / suitを無視した無条件bidにならない
- declaration successが極端に低いままbid率だけ増える状態にならない

## 判定したいこと

最終報告では以下へ明示的に回答する。

1. Napoleon loss `-5 -> 0` でPASS collapseは解消するか
2. PASS top-1 100%は外れるか
3. argmax bidするhandが出現するか
4. 強いhandほどbidする傾向が現れるか
5. bid advantageはどの程度改善するか
6. 成功bid / 失敗bidのadvantageはどうなるか
7. candidate Napoleon率 / declaration success / All Pass率はどう変化するか
8. 無条件bidへの逆collapseは起きるか
9. reward境界が主因と言えるか
10. 次に `-1/-2` 等の中間penaltyを試すべきか、PPO update mechanicsへ進むべきか

## Git / PR運用

最新 `feature/issue-202-non-playing-ai` からIssue専用ブランチを作成する。

PR baseは `feature/issue-202-non-playing-ai`。
main向けPRは作らない。

実装・unit test・smoke・30 iteration実験完了後、Codex review最大4回（review後修正最大3回）を実施する。

blocking / actionable 指摘がなく、必要テスト・実験が成功したら Ready for review にし、親ブランチへmergeする。

mainにはmergeしない。

結果はIssueへコメントし、必要なら `experiments/` 配下へ保存する。

## Acceptance Criteria

- bidding-specific reward v2 を導入
- Napoleon / Napoleon+Adjutant loss reward = 0
- win rewardは #362 と同じ
- non-contract outcomeは0のまま
- bidding relative transformはidentityのまま
- choosing-adjutant / exchange rewardは変更しない
- provenance / resume safetyを更新
- unit testを追加・更新
- smoke fallback=0 / illegal=0
- 30 iteration短期runを実行
- #362とmatched比較
- PASS top-1 / argmax bid / PASS-bestBid gapを確認
- hand-strength別にargmax bidを確認
- target別success / return / advantageを確認
- 無条件bidへの逆collapse有無を判定
- 次のreward/PPO方針を判断できる具体的結論をIssueへ残す

## 非目標

- opponent mix変更
- PPO hyperparameter変更
- network architecture変更
- observation schema変更
- playing policy再学習
- choosing-adjutant / exchange reward変更
- game rule変更
- mainへのMerge
